### ISSUE TITLE
Support on-key-hold shortcuts

### ISSUE THREAD

**Issue by eugenesvk** [lapce/lapce#2906]
## Is your feature request related to a problem? Please describe.

There are a bunch of useful interactions that are blocked by the lack of tap vs. hold differentiation in keyboard shortcuts:

A few examples of what you could get if you had the ability to differentiate:

Convenient same-button mode switching
- tap <kbd>i</kbd> to enter `Insert` mode (in `Normal`)
- tap <kbd>i</kbd> to type `i` (in `Insert`)
- hold <kbd>i</kbd> to cleanly (without any text edits) exit `Insert` mode (in `Insert`)

Adaptive scrolling with a keyboard in Normal mode
- tap <kbd>j</kbd> to go down by 1 line/tap
- short hold < X seconds <kbd>j</kbd> to go down 5 lines / second (or whatever the key repeat rate is)
- long hold > X seconds <kbd>j</kbd> to go down 1 __Page__ / second

## Describe the solution you'd like

Being able to set a time-based shortcut property that would signal whether a key in a shortcut should trigger on tap or on hold

## Describe alternatives you've considered

In principle, you could use some keyboard remapping apps to code this behavior, but they lack any Lapce context (don't know what mode Lapce is in, don't know whether you're in a text field or browsing the menu bar), so that's a major blocker.
Another annoyance is that you don't have info about these keybinds within Lapce, and keeping the shortcuts in sync could be a challenge

## Additional context

N/a
